src/test/FuzzerImpl.cpp

```diff
@@ -918,17 +918,38 @@ struct SponsoredEntryParameters
     int const mSponsorKey; // meaningful only if mSponsored is true
 };
 
+struct ClaimableBalanceParameters : public SponsoredEntryParameters
+{
+    constexpr ClaimableBalanceParameters(int64_t amount, bool sponsored,
+                                         int sponsorKey)
+        : SponsoredEntryParameters(sponsored, sponsorKey), mAmount(amount)
+    {
+    }
+
+    int64_t const mAmount;
+};
+
+auto constexpr MAX_CLAIMABLE_BALANCES_PER_TRUST_LINE = 2;
+
 struct TrustLineParameters : public SponsoredEntryParameters
 {
+    // A claimable balance with a zero "amount" represents an empty entry --
+    // but we use std::array with empty entries rather than std::vector because
+    // the latter can not be constexpr until C++20.
+    using ClaimableBalances = std::array<ClaimableBalanceParameters,
+                                         MAX_CLAIMABLE_BALANCES_PER_TRUST_LINE>;
+
     constexpr TrustLineParameters(AssetID assetID, uint32_t allowTrustFlags,
                                   int64_t assetAvailableForTestActivity,
                                   int64_t spareLimitAfterSetup, bool sponsored,
-                                  int sponsorKey)
+                                  int sponsorKey,
+                                  ClaimableBalances const& claimableBalances)
         : SponsoredEntryParameters(sponsored, sponsorKey)
         , mAssetID(assetID)
         , mAllowTrustFlags(allowTrustFlags)
         , mAssetAvailableForTestActivity(assetAvailableForTestActivity)
         , mSpareLimitAfterSetup(spareLimitAfterSetup)
+        , mClaimableBalances(claimableBalances)
```

Comment:
Also, having them in the trust line avoids having to put the asset and claimant into the `ClaimableBalanceParameters` themselves.  So I think I prefer it this way.  But I'm open to being dissuaded!

Reply:
My main arguments against this arrangement:
1. A claimable balance can exist even if none of its claimants have trust lines for the assets (indeed, it can exist even if the claimant accounts don't exist)
2. An issuer can create a claimable balance even though it won't have a trust line (because it's the issuer)
3. Even ignoring those, which trust line should it be associated with? The sender? One of the claimants? Claimable balances treat all parties uniformly

Reply:
Convincing!  Siddharth also pointed out that we can create claimable balances of native asset, which can't be expressed when they're only referred to from trust lines.  I've factored out `ClaimableBalanceParameters` into their own array, as well as `TrustLineParameters`, and I think the result is dramatically more readable, as well as addressing your arguments and Siddharth's.

I'm not sure, though, whether it would be easier to follow the refactoring and simplification commits before I squash and rebase, or to revisit all the changes directly in their new form after rebasing, so please let me know whether you'd like to do the rebase before you re-review or to wait until after that.

I'll also write up in words the scenarios that I'd like to put into the arrays (tomorrow).

Reply:
Let's squash and rebase at this point. I'll review again once you've done so.

Reply:
Alright, flattening the arrays has allowed dramatic improvements:

- No more `struct` names in the array declarations
- Parameter arrays and operations correspond directly
- Several constants removed
- All setup operations executed using iterators over arrays
- Account ID space can have holes in it if we want
- No more placeholder elements representing "no such element"
- Convenience constructors distinguishing conditions such as the presence or absence of a sponsor or auth flags

I'm working on rebasing that into a new coherent history.